### PR TITLE
Errors and autocorrects when `mixes_in_class_methods` is used without `extend T::Helpers`

### LINKED ISSUE #1805: Sorbet static should error if `mixes_class_methods` is used without `extend T::Helpers`
#### Input

[→ View on sorbet.run](https://sorbet.run/#%23%20typed%3A%20true%0A%0Arequire%20'sorbet-runtime'%0A%0Amodule%20M%0A%20%20%23%20extend%20T%3A%3AHelpers%0A%0A%20%20module%20B%0A%20%20%20%20def%20foo%3B%20end%0A%20%20end%0A%0A%20%20mixes_in_class_methods(B)%0Aend%0A%0Aclass%20A%0A%20%20include%20M%0Aend%0A%0AA.foo)

```ruby
# typed: true

require 'sorbet-runtime'

module M
  # extend T::Helpers

  module B
    def foo; end
  end

  mixes_in_class_methods(B)
end

class A
  include M
end

A.foo
```

#### Observed output

Static:

```
No errors! Great job.
```

Runtime:

```
Traceback (most recent call last):
	1: from file.rb:5:in `<main>'
file.rb:12:in `<module:M>': undefined method `mixes_in_class_methods' for M:Module (NoMethodError)
```

#### Expected behavior

I would expect this to be caught by Sorbet static.

### PR DESCRIPTION
**Pull request by Morriar**
### Motivation

Fixes #1805.

Please ignore the first commit as it is coming from #3381.

### Test plan

See included automated tests.

### CODE REVIEW

File: resolver/resolver.cc [hunk 1]
```diff
@@ -2214,10 +2214,7 @@ class ResolveMixesInClassMethodsWalk {
         }
 
         if (send.args.size() != 1) {
-            if (auto e = ctx.beginError(send.loc, core::errors::Resolver::InvalidMixinDeclaration)) {
-                e.setHeader("Wrong number of arguments to `{}`: Expected: `{}`, got: `{}`",
-                            send.fun.data(ctx)->show(ctx), 1, send.args.size());
-            }
+            // The error will be created later when we'll dispatch the call.
```
Comment (jez) [inline resolver/resolver.cc:2217]:
```suggestion
            // The arity mismatch error will be emitted later by infer.
```

**Review by jez — APPROVED** [review]
This looks good to me but I think that the tests are going to fail because you changed the autocorrect test in #3381

**Comment by Morriar** [comment]
Rebased on master to fix the conflicts with #3381

**Comment by jez** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build result here:

→ https://go/builds/bui_HzzMOFlRETYDZj

**Pull request merged**